File: pset6/caesar.py
import cs50
import sys

if len(sys.argv) != 2:
    print("Sorry, that's not a valid input")
    exit(1)

k = int(sys.argv[1])
print("plaintext: ")
text = cs50.get_string()
cipher = []
print("ciphertext: ")

for letter in text:
    if letter.isupper():
        cipher.append (chr(((ord(letter) - 65 + k) % 26) + 65))
    elif letter.islower:
        cipher.append(chr(((ord(letter) - 97 + k) % 26) + 97))
    else: 
        cipher.append(letter)
        
print("".join(cipher))

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
 
 int main(int argc, string argv[])
 {
    
     if(argc != 2){
         printf("Sorry, that's not a valid input\n");
         return 1;
     }
     
     int k = atoi(argv[1]);
     
     printf("plaintext: ");
     string text = get_string();
     printf("ciphertext: ");
     
     for(int i = 0; i < strlen(text);i++)
     {
         if (islower(text[i]))
         {
             printf("%c", ((text[i]+k-97)%26)+97);
         }
         else if (isupper(text[i])){
             printf("%c", ((text[i]+k-65)%26)+65);
         }
         else printf("%c", text[i]);
     }
     
    printf("\n");
 
     
     
 }